server/src/test/java/org/opensearch/index/mapper/MapperServiceTests.java

```diff
@@ -158,6 +159,21 @@ public void testMappingDepthExceedsLimit() throws Throwable {
         assertThat(e.getMessage(), containsString("Limit of mapping depth [1] has been exceeded"));
     }
 
+    public void testMappingDepthExceedsXContentLimit() throws Throwable {
+        final IllegalArgumentException ex = assertThrows(
+            IllegalArgumentException.class,
+            () -> createIndex("test1", Settings.builder().put(MapperService.INDEX_MAPPING_DEPTH_LIMIT_SETTING.getKey(), 10000).build())
```

Comment:
wdyt about setting the value to `XContentContraints.DEFAULT_MAX_DEPTH + 1` instead of 10000 for this test in case this constant ever changes. For the other test, would it make sense to do the same?